helix-term/src/ui/picker.rs

```diff
@@ -464,6 +517,8 @@ impl<T: 'static> Component for Picker<T> {
             area
         };
 
+        self.set_height(area.height);
```

Comment:
Rather than doing this inside render, you should do this in `required_size`: https://github.com/helix-editor/helix/blob/dfd7dbfcc256a733bd462548f1562a11cd55e32e/helix-term/src/ui/picker.rs#L441-L442

This is how we store the sizing info for popups for example: https://github.com/helix-editor/helix/blob/dfd7dbfcc256a733bd462548f1562a11cd55e32e/helix-term/src/ui/popup.rs#L147-L150

Reply:
I tried this, but that function isn't even called when rendering a `FilePicker`. The documentation of `required_size` says

> *May* be used by the parent component to compute the child area.

(emphasis mine). Also, it seems like `required_size` doesn't return the exact size of the component, but the _minimum_ size. So doing it in `required_size` doesn't work.

Reply:
@archseer Could you take another look on this? I followed your advice and moved `self.set_height` to the `required_size` function. This required some refactoring because `required_size` previously wasn't invoked in `FilePicker`. `FilePicker` uses custom logic in its `render` function to render the picker and the file preview in the center of the terminal, which seems like a hack.

I now changed it so that the `FilePicker` is wrapped in an `Overlay`, a component I created that renders a child component in the center of the terminal. This has the advantage that the child component's `required_size` is called with the correct dimensions.